css/src/atomics/aspect-ratio.scss

```diff
@@ -0,0 +1,17 @@
+$aspect-ratios: (
+	'1-1': '1 / 1',
+	'2-1': '2 / 1',
+	'3-2': '3 / 2',
+	'4-3': '4 / 3',
+	'5-3': '5 / 3',
+	'9-16': '9 / 16',
+	'16-9': '16 / 9'
+) !default;
+
+@each $name, $ratio in $aspect-ratios {
+	.aspect-ratio-#{$name} {
+		aspect-ratio: #{$ratio};
+		width: 100% !important; // must override width to fill container
```

Comment:
Just for my own edification, what happens if we don't include width and height here? My inclination is to not include these rules, but I could be convinced otherwise.

Reply:
It covers some gaps and allows it to scale to the container. So it's mainly there to keep the style agnostic to other style/class dependencies. For instance docs-ui defines width 100% for iframes but no height, so even with the aspect ratio property you get the black bars. Similarly if an image is smaller than the container, it doesn't match the ratio. If its >= to the container width and doesn't have 100% height then it also doesn't match the aspect ratio (notice https://developer.mozilla.org/en-US/docs/Web/CSS/aspect-ratio sets width: auto, height: 100%)

Reply:
After a little testing, I think that we cannot go the route of placing `height: 100%` on this, because this implies the height of a parent element. You can test the undesirable behavior by moving the video element out of its current spot (using dev tools) and putting it as a child of a tall parent element. In this case, I did this with the parent element being `#article`. Things get very tall very fast. 

<img width="796" alt="image" src="https://github.com/microsoft/atlas-design/assets/30843002/522a7750-244f-44fb-9779-7d499446a357">

Good news is I'm not sure we need it. Aspect ratio seems to work fine without it, if we apply an object-fit property as well. I think we _could remove height_ too because our reset applies `max-width: 100%` to images and videos. That said, I think we may want to consider apply `max-width: 100%` for safety, and so that other elements can get the same.

Reply:
Cancel that thing about width, think we _do_ need width

Reply:
@wibjorn after more testing and reflecting on the use case that brought this proposal up I think we should keep it a true atomic without a width or height. I've removed those from the style.

Thoughts behind this are:
1. The iframe video on the homepage had black bars because a width/height attribute were set on the iframe element. Without those attributes set the `aspect-ratio` style resolves the issue.
2. Images that are less than 100% width don't maintain the ratio but a majority of our images are larger than their containers, and we also have an `image` class that sets the img width to 100% which when used with this atomic would allow the ratio to be applied to smaller images.
3. This keeps it more inline with the principle of atomics and other classes that target width or object fit can be added for special use cases.
4. The width/height 100% !important was mainly to force the ratio on an element even if a different width or height are applied with attributes or inline styles, but that feels out of scope of an atomic.

Reply:
Great! That was my initial inclination as well.